[Jem Zhang] iP

AppleJem opened the pull request:
# Donk

> Damn bruh you be having lots to do, why not track them with DONK 🙄

Donk frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

**And it is FREE!**

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

AppleJem committed:
style: Changed chatbot name to Donk

AppleJem committed:
feat: Added chatbot greeting

AppleJem committed:
feat: echo and bye

AppleJem committed:
feat: list feature

AppleJem committed:
feat: added Task class. Unmark and mark tasks

AppleJem committed:
feat: add tasks of todo, deadline, event. minor refactor

AppleJem committed:
feat: Added testfile and expected file

AppleJem committed:
feat: basic error handling

AppleJem committed:
feat: delete function

AppleJem committed:
Add feature to save and read file

AppleJem committed:
add datetime parser to deadline and event

AppleJem committed:
refactor ui, storage, tasklis

AppleJem committed:
change some system out messages to use ui class

AppleJem committed:
divide classes into packages

AppleJem committed:
Merge branch 'add-gradle-support' of https://github.com/AppleJem/cs2103t-ip
Pull gradle config files from remote

AppleJem committed:
add tests for storage and parser

AppleJem committed:
configure for building jar with gradle shadowjar

AppleJem committed:
add docstrings for most methods

AppleJem committed:
add search function

AppleJem committed:
fix parser to read m/d/yyyy format without time

AppleJem committed:
add test for search function in ParserTest

AppleJem committed:
fix bug where error message not displaying for wrong format event and todo tasks

jayjay19630 commented on src/main/java/donk/Parser.java:
```diff
@@ -0,0 +1,142 @@
+     * - "delete <index>": Deletes the task at the given index.
+     * - "todo <description>": Adds a new todo task with the given description.
+     * - "deadline <description> /by <date>": Adds a new deadline task with the given description and due date.
+     * - "event <description> /start <start> /end <end>": Adds a new event task with the given description, start time, and end time.
```
I like the detailed documentation for your parser!

jayjay19630 commented on src/main/java/donk/task/TaskList.java:
```diff
@@ -0,0 +1,74 @@
+     *
+     * @param tasks The list of tasks to initialize the TaskList with.
+     */
+    public TaskList (List<Task> tasks) {
```
Since this constructor is a method as well, please avoid whitespaces between method names and arguments, as per the coding standard.

<img width="981" alt="Screenshot 2024-09-03 at 11 57 55 AM" src="https://github.com/user-attachments/assets/156dc2bf-6bd9-42ba-87e9-a9460da47ccb">

jayjay19630 commented on src/test/java/donk/StorageStub.java:
```diff
@@ -0,0 +1,11 @@
+package donk;
+
+
+public class StorageStub extends Storage {
+
+    public StorageStub(String filePath) {
+        super(filePath);
+    }
+
+
+}
```
I like your use of a stub to avoid dependencies!

jayjay19630 commented on src/test/java/donk/task/ToDoTest.java:
```diff
@@ -0,0 +1,10 @@
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+public class ToDoTest {
+    @Test
+    public void dummyTest() {
+        assertEquals(2, 2);
+    }
```
Although this file might still be in progress, please don't forget to remove this unrelated dummy test when you are finished.

AppleJem replied:
Good spot! Thanks friend :D

jayjay19630 commented on src/main/java/donk/Parser.java:
```diff
@@ -0,0 +1,142 @@
+                tasks.add(t);
+
+            }
+            System.out.println("    Got it. I've added this task:");
+            System.out.println("        " + t.toString());
+            System.out.println("    You now have " + tasks.size() + " tasks");
```
Since we have an existing `Ui.java` class for printing to the console, should we move into `Ui.java` for consistency?

jayjay19630 commented on src/main/java/donk/Ui.java:
```diff
@@ -0,0 +1,106 @@
+    /**
+     * Greets the user with a welcome message when the application starts.
+     */
+    public void greet() {
+        String greetMsg = " ____________________________________________________________\n" +
+                " Hello! I'm donk.Donk, the super intelligent chatbot\n" +
+                " What can I do for you?\n" +
+                "____________________________________________________________\n";
+
+        System.out.println(greetMsg);
+    }
```
I noticed that all `Ui.java` methods do not require instantiated variables inside the class, so technically Ui does not even have to be instantiated as an object.  Do you think it would be better make `Ui.java` methods static instead? 

An implication of this would be that Ui no longer has to be stored as an object inside `Donk.java`. Instead, these methods can just be called by doing `Ui.parse(...)`, `Ui.bye()`, etc.

AppleJem replied:
That's a good point!

AppleJem commented:
# Donk

> Damn bruh you be having lots to do, why not track them with DONK 🙄

Donk frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

**And it is FREE!**

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

AppleJem committed:
correct styles in all files using checkstyles

AppleJem committed:
Merge tag 'A-CheckStyle'

Style correction and changes

kushaldas27 commented on src/main/java/donk/Parser.java:
```diff
@@ -0,0 +1,155 @@
+                tasks.add(t);
+
+            }
+            System.out.println("    Got it. I've added this task:");
+            System.out.println("        " + t.toString());
+            System.out.println("    You now have " + tasks.size() + " tasks");
```
Consider using "\t" to insert a tab.

AppleJem replied:
Oh I did not know that hahaha, thanks!

kushaldas27 commented on src/main/java/donk/Ui.java:
```diff
@@ -0,0 +1,112 @@
+     * Greets the user with a welcome message when the application starts.
+     */
+    public void greet() {
+        String greetMsg = " ____________________________________________________________\n"
+                + " Hello! I'm donk.Donk, the super intelligent chatbot\n"
+                + " What can I do for you?\n"
+                + "____________________________________________________________\n";
```
Consider creating a drawLine() function that returns the separating line as a String. Would help cut down on a lot of repeated code.

kushaldas27 commented on src/main/java/donk/Storage.java:
```diff
@@ -0,0 +1,92 @@
+        while (s.hasNext()) {
+            String line = s.nextLine();
+            String[] data = line.split("\\|");
+            if (line.charAt(0) == 'T') {
+                tasks.add(new ToDo(data[2]));
+            } else if (line.charAt(0) == 'D') {
+                tasks.add(new Deadline(data[2], data[3]));
+            } else if (line.charAt(0) == 'E') {
+                tasks.add(new Event(data[2], data[3], data[4]));
```
Consider using a switch statement since you have multiple cases.

AppleJem replied:
Good point!

AppleJem committed:
add GUI that just echoes user message for now

AppleJem committed:
add input parsing to gui

AppleJem committed:
Merge branch 'branch-level-10'

Add GUI chat bot interface

AppleJem committed:
add assert statements to check ui and storage parser

Ensure that ui and storage objects are not equal to null
which would cause a problem when calling methods on it

AppleJem committed:
refoctor command execution to Donk instead of Parser

refactor to conform to length and prevent excessive nesting

AppleJem committed:
Refactor mark, unmark, delete funtions

to reduce length of execute function in Donk

AppleJem committed:
refactor todo, deadline, event, find commands

AppleJem committed:
fix parseIndex to check that index is provided

AppleJem committed:
Merge pull request #2 from AppleJem/branch-A-CodeQuality

Branch a code quality

AppleJem committed:
Merge branch 'master' of https://github.com/AppleJem/cs2103t-ip

pull changes from merged A-CodeQuality branch

AppleJem committed:
fix missing return statement on bye

AppleJem committed:
add sorted list feature

Command 'list sorted' returns tasks sorted by date
Task comparison done by adding compareTo method, which
compares ISO string of datetime of task
This allows us to sort by date

AppleJem committed:
Merge pull request #3 from AppleJem/branch-C-sort

Branch c sort

AppleJem committed:
add user guide and screenshot

AppleJem committed:
Merge branch 'master' of https://github.com/AppleJem/cs2103t-ip

merging from remote repo

AppleJem committed:
change width specification of image in readme

AppleJem committed:
add title bar and update userguide screenshot

AppleJem committed:
change erroneous parser function input in test

AppleJem committed:
change datetime parsing in Event. Update error message

AppleJem committed:
Merge branch 'branch-fix-datetime'

Merging fixed event datetime parsing

AppleJem committed:
edit test cases to adjust to refactor

AppleJem committed:
edit build.gralde file to include java fx dependencies

AppleJem committed:
Update README.md

AppleJem committed:
Update README.md

AppleJem committed:
Update README.md

AppleJem committed:
Update README.md
